pkg/florist/netstat.go

```diff
@@ -113,3 +113,28 @@ func PublicIPs() ([]string, error) {
 
 	return ips, nil
 }
+
+// PrivateIP returns the first matching IP that belongs to the given network CIDR.
+func PrivateIP(cidr string) (string, error) {
+	_, network, err := net.ParseCIDR(cidr)
+	if err != nil {
+		return "", fmt.Errorf("network CIDR: %s: error: %s", cidr, err)
+	}
+
+	ips, err := PrivateIPs()
+	if err != nil {
+		return "", err
+	}
+
+	for _, ip := range ips {
+		ip := net.ParseIP(ip)
```

Comment:
I had missed this one, super confusing, it seems that the type of `ip` is changing (because the scope is changing, obviously).

We can call the internal ip `addr`.